### PR TITLE
Makes crossbows buildable

### PR DESCRIPTION
**Pull request by jakmak6**
Adds the crossbow frame to the construction options of wood planks for a price of 5 planks. With 3 metal rods, a welder, 3 cable coils, 3 sheets of plastic, 3 more cable coils and a screwdriver you too can have your very own vox grade crossbow (powercell sold separate).

crossbow

Assembled from left to right starting with wood and ending with a screwdriver to produce a functional crossbow. If requested i can change the required parts/steps for construction

If merged i will add it to the wiki

Credit to baystation12 for the sprites and idea

* rscadd: Crossbows can now be built from wood planks

### CODE REVIEW

**Comment by Bpage5** [comment]
I feel making this crossbow require plastic to make would relegate it to the same tier of usefulness as the venom sword, but considering what the voxbow does, I can't really think of a better way to make it hard to get.

**Comment by Kurfursten** [comment]
How about reversible construction for field-stripping bows?

It would be pretty slick to disarm someone and screwdriver it for the second it's on the ground

**Comment by ghost** [comment]
These pretty much suck, don't they?

**Comment by nuklearcellphoneg** [comment]
Style > Powergaming

**Comment by Exxion** [comment]
@Unknown-as-Captain It's been a long time since I've used one, but if memory serves, they are very powerful for one shot, but then take so long to reload  you'd just get killed unless you have support
So like a worse pneumatic cannon

**Comment by Shadowmech88** [comment]
I'm not against making it require plastic, but does it really make sense for the crossbow's *arms* to be made of plastic?

**Comment by PJB3005** [comment]
@Shadowmech88 I mean what else would you use in the context of SS13? Plastic is already inconsistent so it's not too far off.

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 1]
```diff
@@ -212,3 +212,93 @@
 		arrow = null
 		tension = 0
 		icon_state = "crossbow"
+
+
+
+/* construction*/
+/obj/item/weapon/crossbowframe
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/bow.dm]: This has no reason to be `/obj/item/weapon`, make it `/obj/item`

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 2]
```diff
@@ -212,3 +212,93 @@
...
+	icon_state = "crossbowframe0"
+	item_state = "crossbow-solid"
+
+	var/buildstate = 0
+
+/obj/item/weapon/crossbowframe/update_icon()
+	icon_state = "crossbowframe[buildstate]"
+
+/obj/item/weapon/crossbowframe/examine(mob/user)
+	. = ..(user)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/bow.dm]: `user` is implicitly passed down, you can remove the argument.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Make the entire construction process use [construction datums](https://github.com/vgstation-coders/vgstation13/wiki/Construction-Datums%3A-%3F%3F%3F-and-a-How-To)

**Comment by Blithering** [comment]
Neat that it's buildable, but as it stands its a lot cheaper and easier to make and use than a blunderbuss, which fills the same niche. Adding more steps and materials to the construction would help I guess.

**Comment by PJB3005** [comment]
1. Make blunderbusses
2. fill them to the brim with welder fuel.
3. Throw them around.
4. Watch as assistants blow themselves up.

**Pull request merged**